src/main/java/javaslang/λ.java

```diff
@@ -173,7 +105,103 @@ public String toString() {
          *
          * @return the parameter types
          */
-        Class<?>[] parameterArray();
+        Class<?>[] parameterTypes();
```

Comment:
The array is intentionally not cloned/duplicated, i.e. it may be modified. However, each new call of `getType()` will create a new parameterTypes array.

Mutation is generally discouraged (by me). Don't do that - or at most carefully.